app/models/task.py

```diff
@@ -4,3 +4,27 @@
 
 class Task(db.Model):
     task_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable = True)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"), nullable = True)
+
+    def task_dict(self): 
+        return {
+            "id": self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at is not None
+        }
+
+
+    def tasked_dict(self):
+        return {
+            "id": self.task_id,
+            "goal_id": self.goal_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at is not None
+        }
+
```

Comment:
Great job using helper methods Veronica!!!

Reply:
The name of the second helper method is a little confusing, why is a task with a goal_id considered "tasked" vs. "task"? "task_with_goal_id_dict" would be an example of a descriptive name for this method.

Reply:
Regardless, we can actually combine these two methods! Instead of writing a completely new method to return a version of the task dictionary with a `goal_id`, we can use a conditional to add the `goal_id` if it exists for a task:

```suggestion
    def task_dict(self): 
       task_dict = {
            "id": self.task_id,
            "title": self.title,
            "description": self.description,
            "is_complete": self.completed_at is not None
        }
        
       if self.goal_id:
           task_dict["goal_id"] = self.goal_id
       
       return task_dict
```